lib/client.js

```diff
@@ -902,11 +905,16 @@ SMTPClient.prototype._actionAUTHComplete = function(str) {
         try {
             response = str.split(' ');
             response.shift();
-            response = JSON.parse(new Buffer(response.join(' '), 'base64').toString('utf-8'));
-
-            if ((!this._xoauth2.reconnectCount || this._xoauth2.reconnectCount < 200) && ['400', '401'].indexOf(response.status) >= 0) {
+            responseString = new Buffer(response.join(' '), 'base64').toString('utf-8')
+            response = JSON.parse(responseString);
+            if (+response.status === 400 && response.schemes === 'Bearer') {
+                this._onError(new Error('Invalid login - Access token expired: ' + responseString), 'AuthError', response);
+                return;
+            } else if ((!this._xoauth2.reconnectCount || this._xoauth2.reconnectCount < 200) && ['400', '401'].indexOf(response.status) >= 0) {
```

Comment:
This else block will be removed once I can confirm this branch of logic is not encountered in production use